pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -0,0 +1,502 @@
+package awskms
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/aws/aws-sdk-go-v2/service/kms/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/pkg/server/plugin/keymanager"
+	"github.com/spiffe/spire/proto/spire/common/plugin"
+	"github.com/zeebo/errs"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	pluginName       = "awskms"
+	aliasPrefix      = "alias/"
+	defaultKeyPrefix = "SPIRE_SERVER_KEY/"
+
+	keyIDTag = "key_id"
+	aliasTag = "alias"
+)
+
+var (
+	kmsErr = errs.Class(pluginName)
```

Comment:
Agreed on errors and fmt. The grpc status code is another story. I considered adding them early in the development, but I noticed that SPIRE Server doesn't make use of them, and I was not able to find other plugins implemented that way either.

Reply:
The history of errors produced by plugins is not every regimented. We originally just returned manually formatted errors. At some point early on we introduced the `errs` package so we could get auto-tagged errors (i.e. `errs.Class`). However, errors produced via the gRPC `status` package are preferred because we can attach a status code to them that is nice for the logging of these errors by SPIRE core. Otherwise, we get an `Unknown` for everything, which isn't very helpful. It can still be helpful to prefix the errors with the name of the plugin. So a helper something like the following might be useful:

```
func newError(code codes.Code, msg string) error {
    return status.Error(code, pluginName + ": " + msg)
}

func newErrorf(code codes.Code, format string, args ...interface{}) error {
    return status.Error(code, pluginName + ": " + fmt.Sprint(format, args...))
}
```